Source/DafnyCore/Verifier/BoogieGenerator.Methods.cs

```diff
@@ -1461,6 +1461,9 @@ public partial class BoogieGenerator {
       // Note, it is as if the trait's method is calling the class's method.
       var contextDecreases = overryd.Decreases.Expressions;
       var calleeDecreases = original.Decreases.Expressions;
+      var T = (TraitDecl)(overryd is Function f1 ? f1.EnclosingClass : ((Method)overryd).EnclosingClass);
+      var I = (TopLevelDeclWithMembers)(original is Function f2 ? f2.EnclosingClass : ((Method)original).EnclosingClass);
```

Comment:
Instead of the `? :`, cast to `MemberDecl` and then take its `.EnclosingClass`. Like this:

``` c#
((MemberDecl)overryd).EnclosingClass
```

Reply:
Thanks I tried for some time, but not hard enough, as it turns out.